app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx

```diff
@@ -0,0 +1,271 @@
+
+    this.height = container.offsetHeight;
+    this.width = container.offsetWidth;
+    this.noAvailableData = false;
+    const defectTypesChartData = {};
+    const defectTypesChartDataOrdered = [];
+    const defectTypesChartColors = {};
+    const values = widget.content.result[0].values;
+    Object.keys(values).forEach((key) => {
+      if (key.indexOf('$executions$') === -1) {
+        const defectStatus = key.split('$')[2].toUpperCase();
+        defectTypesChartData[key] = +values[key];
+        defectTypesChartColors[key] = COLORS[`COLOR_${defectStatus}`];
+      }
+    });
+
+    widget.contentParameters.contentFields.forEach((field) => {
+      Object.keys(defectTypesChartData).forEach((key) => {
+        if (field === key) {
+          defectTypesChartDataOrdered.push([field, defectTypesChartData[field]]);
+        }
+      });
+    });
+
+    if (!defectTypesChartDataOrdered.length) {
+      this.noAvailableData = true;
+      return;
+    }
+
+    this.rightChartHidden = !defectTypesChartDataOrdered.length;
+    this.defectItems = getDefectItems(defectTypesChartDataOrdered);
+    this.issueConfig = {
+      data: {
+        columns: defectTypesChartDataOrdered,
+        type: 'donut',
+        order: null,
+        colors: defectTypesChartColors,
+      },
+      interaction: {
+        enabled: !self.isPreview,
+      },
+      padding: {
+        top: self.isPreview ? 0 : 85,
+      },
+      legend: {
+        show: false, // we use custom legend
+      },
+      donut: {
+        title: 0,
+        label: {
+          show: false,
+          threshold: 0.05,
+        },
+      },
+      tooltip: {
+        grouped: false,
+        position: this.getIssuesPosition,
+        contents: this.renderIssuesContents,
+      },
+      onrendered: () => {
+        let total = 0;
+        if (this.issuesChart) {
+          this.issuesChart.data.shown().forEach((dataItem) => {
+            total += dataItem.values[0].value;
+          });
+          this.issuesNode.querySelector('.c3-chart-arcs-title').childNodes[0].textContent = total;
+        }
+      },
+    };
+
+    this.setState({
+      isConfigReady: true,
+    });
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  getIssuesPosition = (d, width, height) => {
+    const rect = this.issuesNode.getBoundingClientRect();
+    const left = this.x - rect.left - width / 2;
+    const top = this.y - rect.top - height;
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  resizeIssuesChart = () => {
+    const newHeight = this.props.container.offsetHeight;
+    const newWidth = this.props.container.offsetWidth;
+    if (this.height !== newHeight) {
+      this.issuesChart.resize({
+        height: newHeight,
+      });
+      this.height = newHeight;
+    } else if (this.width !== newWidth) {
+      this.issuesChart.flush();
+      this.width = newWidth;
+    }
+  };
+
+  renderIssuesContents = (d, a, b, color) => {
```

Comment:
What is `d, a, b`? Please avoid to name variables with one symbol.

Reply:
Unfortunately this function is from d3, and the unused items are actually called a and b and I have no idea what they are for. But they need to be there as arguments so I can use color and the first argument (which I can rename to "data" or something)